Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,14 +48,58 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            pypircPath = tl.getVariable("PYPIRC_PATH");
+            const pypirc = fs.readFileSync(pypircPath, 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let reposList: string[] = [];
+
+            for (let entry of newEndpointsToAdd) {
+
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
+
+                if (entry.packageSource.feedName in fileContent){
+                    tl.warning(tl.loc("DuplicateRegistry", entry.packageSource.feedName));
+                    if (internalFeed.includes(entry)) {
+                        internalFeed.pop();
+                        continue;
+                    }
+                    externalEndpoints.pop();
+                    continue;
+                }
+
+                let repo = new Repository(
+                    entry.packageSource.feedName,
+                    entry.packageSource.feedUri,
+                    entry.username,
+                    entry.password
+                );
+
+                reposList.push(repo.toString() + `${os.EOL}`);
```

Comment:
IF the EOL is added to the tostring method above - should this be removed? or is two EOL intentional?

Reply:
It's intentional-- 2 EOLs creates a space between authenticated repos so it's easier to read